packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -41,8 +42,8 @@ export async function fetchSnapshot(
     versionId: string,
     fetchFullSnapshot: boolean,
     logger: ITelemetryLogger,
-    snapshotDownloader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
-): Promise<IOdspResponse<IOdspSnapshot>> {
+    snapshotDownloader: <T>(url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<T>>,
+): Promise<ISnapshotCacheValueV2> {
```

Comment:
I'd not tie the response of fetching to caching. You can define almost identical interface but without version and use it for most of the communication (including here), but only add version right before adding it into the cache. Same with reading - version is only required by version layer, beyond that there is no need to leak that detail.

Reply:
Removed it.